include/mrdox/Metadata/Type.hpp

```diff
@@ -13,13 +14,218 @@
 #define MRDOX_API_METADATA_TYPE_HPP
 
 #include <mrdox/Platform.hpp>
+#include <mrdox/Metadata/Specifiers.hpp>
 #include <mrdox/Metadata/Symbols.hpp>
+#include <mrdox/MetadataFwd.hpp>
+#include <memory>
 #include <string>
 #include <string_view>
+#include <vector>
 
 namespace clang {
 namespace mrdox {
 
+enum class QualifierKind
+{
+    None = 0,
+    Const = 1,
+    Volatile = 2
+};
+
+QualifierKind operator~(QualifierKind& a) noexcept;
+QualifierKind operator&(QualifierKind a, QualifierKind b) noexcept;
+QualifierKind operator&=(QualifierKind& a, QualifierKind b) noexcept;
+QualifierKind operator|(QualifierKind a, QualifierKind b) noexcept;
+QualifierKind operator|=(QualifierKind& a, QualifierKind b) noexcept;
+QualifierKind operator^(QualifierKind a, QualifierKind b) noexcept;
+QualifierKind operator^=(QualifierKind& a, QualifierKind b) noexcept;
```

Comment:
... the heck is all this?

Reply:
I'm not particularly happy with this either.. it will go away

Reply:
only 3 values in the enum yet 7 functions to operate on them... lol